File: HelpForTeacher (Java)/app/src/main/java/com/teacherhelper/MainActivity.java
package com.teacherhelper;

import android.annotation.SuppressLint;
import android.content.Intent;
import android.support.v7.app.ActionBar;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.os.Handler;
import android.view.MotionEvent;
import android.view.View;

import com.example.lili.teacherhelper.R;
import com.kinvey.android.Client;

/**
 * An example full-screen activity that shows and hides the system UI (i.e.
 * status bar and navigation/system bar) with user interaction.
 */
public class MainActivity extends AppCompatActivity {
    private Client client;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        setContentView(R.layout.activity_main);
        client = ((TeacherApp)getApplication()).getSharedClient();x`
        final Handler handler = new Handler();
        handler.postDelayed(new Runnable() {
            @Override
            public void run() {
                if(client.user().isUserLoggedIn()){
                    Intent intent = new Intent(MainActivity.this, ProfileActivity.class);
                    startActivity(intent);
                } else {
                    Intent intent = new Intent(MainActivity.this, LoginActivity.class);
                    startActivity(intent);
                }
            }
        }, 1000);
    }
}

File: HelpForTeacher (Kotlin)/app/src/main/java/com/teacherhelper/MainActivity.kt
package com.teacherhelper

import android.content.Intent
import android.os.Bundle
import android.os.Handler
import android.support.v7.app.AppCompatActivity
import com.example.lili.teacherhelper.R
import com.kinvey.android.Client
import com.kinvey.java.User

/**
 * An example full-screen activity that shows and hides the system UI (i.e.
 * status bar and navigation/system bar) with user interaction.
 */
class MainActivity : AppCompatActivity() {
    private var client: Client? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(R.layout.activity_main)
        client = (application as TeacherApp).sharedClient!!
        val handler = Handler()
        handler.postDelayed({
            if (client!!.user<User<*>>().isUserLoggedIn()) {
                val intent = Intent(this@MainActivity, ProfileActivity::class.java)
                startActivity(intent)
            } else {
                val intent = Intent(this@MainActivity, LoginActivity::class.java)
                startActivity(intent)
            }
        }, 1000)
    }
}
